hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/ClasspathXmlJetConfig.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.config;
+
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.jet.impl.config.XmlJetConfigBuilder;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+import static com.hazelcast.util.Preconditions.checkTrue;
+
+/**
+ * A {@link JetConfig} which is initialized by loading an XML configuration file from the classpath.
+ *
+ * @see FileSystemXmlJetConfig
+ * @see UrlYamlJetConfig
+ * @see InMemoryYamlJetConfig
+ */
+public class ClasspathXmlJetConfig extends JetConfig {
```

Comment:
I'm not sure if we need this class? I prefer the static methods we have like in `JetConfig.loadFromClasspath`. It's easier to discover them. 
I don't like the idea of extending `JetConfig`. The same applies for the other versions of this class. We deliberately didn't offer these for Jet, and instead only mainly offer `loadFromStream()` which covers more cases.

Reply:
The approach that you are suggesting makes implementation more complicated around the `JetConfig` class since each option has to work it's way around whether it is loading XML or YAML. For example, in the `loadFromStream()`  we don't know whether it is a XML or YAML stream.

Reply:
we could have loadXmlFromStream and so on, and deprecate the current methods?